OCPI 2.3.0

reinierl opened the pull request:
Creating a pull request as a place where reviewers can see a diff of changes in OCPI 2.3.0 compared to the latest OCPI 2.2.1 specification text.

reinierl committed:
fix tariff calculation table for an example

reinierl committed:
fix grammar

reinierl committed:
fix broken table syntax in commands module

reinierl committed:
add vehicle type to EVSE object

reinierl committed:
replace dead eMI3 links by archive.org links

reinierl committed:
add requirement to have a photo of EVSE for people w disabilities

reinierl committed:
update introduction for OCPI 2.3.0

reinierl committed:
explicitly allow using unspecified JSON field names

reinierl committed:
add explanation of class, enum and openenum types

reinierl committed:
open up enums that people may want to extend

reinierl committed:
update vehicle types truck classification

reinierl committed:
add support for N-Am taxes to Tariffs

as directed by Chargehub's document.

reinierl committed:
allow listing multiple taxes in Session/CDR costs

according to proposal from ChargeHub.

reinierl committed:
make min_price/max_price work with the new Price type

reinierl committed:
allow OCPI 3.0 style presentation of multitenant Platforms as Hubs

reinierl committed:
add useful new enum values from OCPI 3.0

reinierl committed:
update changelog and version history for 2.3.0

reinierl committed:
update version to 2.3.0-rc1

reinierl committed:
add EMPLOYEES value for ParkingRestriction

madman-81 commented on mod_locations.asciidoc:
```diff
@@ -1086,23 +1092,23 @@ The category has to be set accordingly to the image content in order to guarante
+|PLUGGED   |Parking is only allowed while plugged in (charging).
 |CUSTOMERS |Parking spot for customers/guests only, for example in case of a hotel or shop.
-|MOTORCYCLES |Parking spot only suitable for (electric) motorcycles or scooters. 
+|TAXI      |Parking only for taxi vehicles.
```
Who not use the plural form here like (most) others?

madman-81 commented on mod_tariffs.asciidoc:
```diff
@@ -918,8 +959,8 @@ A Price Component describes how a certain amount of a certain dimension being co
-|price |<<types.asciidoc#types_number_type,number>> |1 |Price per unit (excl. VAT) for this dimension.
-|vat |<<types.asciidoc#types_number_type,number>> |? |Applicable VAT percentage for this tariff dimension. If omitted, no VAT is applicable. Not providing a VAT is different from 0% VAT, which would be a value of 0.0 here.
+|price |<<types.asciidoc#types_number_type,number>> |1 |Price per unit for this dimension. This is including or excluding taxes according to the `tax_included` field of the Tariff that this PriceComponent is contained in.
+|vat |<<types.asciidoc#types_number_type,number>> |? |Applicable VAT percentage for this tariff dimension. If omitted, no VAT is applicable. 
```
Shouldn't this support different types of taxes as well? See also type Price. If not, it this the total percentage of taxes or just the VAT?

madman-81 commented on types.asciidoc:
```diff
@@ -63,8 +90,21 @@ Unless mentioned otherwise, numbers use 4 decimals and a _sufficiently large amo
+[cols="3,2,1,10",options="header"]
+|===
+|Property       |Type                         |Card. |Description
+|name           |<<types_string_type,string>> |1     |A description of the tax. In countries where a tax name is required like Canada this can be something like "QST". In countries where this is not required, this can be something more generic like "VAT" or "General Sales Tax".
```
It would be nice to make this an (open)Enum so that there is a list of known taxes

madman-81 commented on types.asciidoc:
```diff
@@ -2,6 +2,33 @@
+In the serialized JSON form of OCPI messages, enum values are serialized as JSON strings.
+
+[[types_openenum_type]]
+=== OpenEnum _type_
```
Having the data types specified as OpenEnum makes it more difficult to ensure interoperability between parties. Although I understand the need for more flexibility, there is also the risk of getting different values with (more or less) the same meaning. And the risk of losing track of the meaning of a value in the first place. 

An alternative solution might be a DynamicEnum that is still a predefined set of possible values, but with a central list of accepted values. This list could be here on GitHub and if parties feel the need for extra values, they can create a pull request to add a new value. This way there is no need for a new release yet still a centralised list of accepted values.

woutstevens1 commented on mod_locations.asciidoc:
```diff
@@ -731,6 +731,8 @@ The capabilities of an EVSE.
 |CONTACTLESS_CARD_SUPPORT |EVSE has a payment terminal that supports contactless cards.
 |CREDIT_CARD_PAYABLE |EVSE has a payment terminal that makes it possible to pay for charging using a credit card.
 |DEBIT_CARD_PAYABLE |EVSE has a payment terminal that makes it possible to pay for charging using a debit card.
+|ISO_15118_2_PLUG_AND_CHARGE |The EVSE supports authentication of the Driver using a contract certificate stored in the vehicle according to ISO 15118-2.
+|ISO_15118_20_PLUG_AND_CHARGE |The EVSE supports authentication of the Driver using a contract certificate stored in the vehicle according to ISO 15118-20.
```
What about `ISO_15118_20_BIDIRECTIONAL_CHARGING`?

woutstevens1 commented on version_information_endpoint.asciidoc:
```diff
@@ -170,7 +170,7 @@ If there are dependencies between modules, it will be mentioned in the affected
-==== VersionNumber _enum_
+==== VersionNumber _OpenEnum_
 
 List of known versions.
```
OCPI 2.3.0 is missing from the `VersionNumber` options

reinierl committed:
make TAXIS parking restriction plural for consistency

reinierl committed:
add separate, referenced parking objects in Location objects

reinierl committed:
adapt examples for introduction of Parking class

reinierl committed:
add physical_reference for parking spaces

reinierl committed:
fix Tariffs examples for before_taxes/after_taxes field names

reinierl committed:
make ISO 15118 capability indicators appear on the Connector object

reinierl committed:
add field for list of accepted eMSPs

This is a rather daft feature request according to most, but as it is found in
law legislated by the European Commission, we don't really have a choice.

reinierl committed:
add 2.3.0 as VersionNumber option

reinierl committed:
update changelog and version history with changes since rc1

reinierl committed:
change version to 2.3.0-rc2

ahzf commented on mod_locations.asciidoc:
This should be a list of the **EMSP_Ids** of the eMSPs offering contract-based payment options that are accepted at this EVSE.

"names" would only lead to 1024 different names for the same EMSP.

reinierl replied:
the European Commission wants "Information indicating the name of those mobility service providers that are offering contract-based payment options and are accepted in a recharging station" and they can have it. There are lots of arguments you can put forward about how to make this more practical but we're just trying to do the minimum necessary thing to comply with their rules for how to communicate with National Access Points in Europe.

ahzf replied:
COMMISSION DELEGATED REGULATION (EU) …/...
of XXX
amending Regulation (EU) 2023/1804 of the European Parliament and of the Council as
regards additional data types on alternative fuels infrastructure
https://ec.europa.eu/info/law/better-regulation/have-your-say/initiatives/14347-Data-on-alternative-fuels-infrastructure-additional-data-types_en page 5:

> Article 1 Paragraph 2 of Article 20 of Regulation (EU) 2023/1804 is replaced by the following: 
> [...]
> (b) (7) Mobility service providers offering contract-based recharging

So they do not specify a `name` here. Compared to data sets where they clearly specify an `id` here it's all up to the implementer.

ahzf commented on transport_and_format.asciidoc:
```diff
@@ -371,6 +370,17 @@ include::examples/transport_and_format_get_token_list_example.json[]
 }
 ----
 
+==== Non-specified JSON fields
```
This are two very different things:
- You do not actively reject additional JSON properties
- You coordinate those extension-properties across the industry

For the later it is all up to you to use e.g. this repository to inform about your extension ideas, the refine it, to make other use it and finally to standardize it within one of the next OCPI versions.

reinierl committed:
note that accepted_service_providers is for NAPs only

Sikredo committed:
add payments module

reinierl committed:
promote Parking from class to object

this way we can give it a place in the class diagram

reinierl committed:
remove vehicle_types field from EVSE

it is now in the Parking object

reinierl committed:
require people to look at EVRF's extension processes when extending OCPI

reinierl committed:
move EVSE position from Parking to EVSEParking join type

reinierl committed:
make Parking object optional

ie make it possible to share Locations that do not contain any Parking objects.

reinierl committed:
explain Parking object, note it is typically not needed

reinierl committed:
add preauthorize_amount field to Tariff

this removes the need to abuse max_price for this purpose.

reinierl committed:
add support phone number to Location object

Sikredo committed:
fix typos and missing sentences in Payments module

reinierl committed:
update version to 2.3.0-rc3

Sikredo committed:
add unece code mapping to vehicle-types

Sikredo committed:
fix typo

reinierl committed:
add clarifying statement about column of UNECE codes

reinierl committed:
update references to eMI3 formats to go via evroaming.org webpage

reinierl committed:
version 2.3.0-rc4

benvangameren committed:
version 2.3.0
